mod/01_core/set/all/initialize.rb

```diff
@@ -1,13 +1,24 @@
-JUNK_INIT_ARGS = %w{ missing skip_virtual id }
+JUNK_INIT_ARGS = %w{ find_unused_name missing skip_virtual id }
 
 module ClassMethods
   def new args={}, options={}
     args = (args || {}).stringify_keys
+    args['name'] = find_unused_name args['name'] if args['find_unused_name']
     JUNK_INIT_ARGS.each { |a| args.delete(a) }
     %w{ type type_code }.each { |k| args.delete(k) if args[k].blank? }
     args.delete('content') if args['attach'] # should not be handled here!
     super args
   end
+
+  def find_unused_name base_name
+    test_name = base_name
+    add = 1
+    while Card.exists?(test_name) do
+      test_name = "#{base_name}#{add}"
+      add +=1
+    end
+    test_name
+  end
 end
```

Comment:
autoname seems like the wrong name for this.  Does it do that same thing, and where it is?  Seemed to me that was an old feature that doesn't quite make sense now.  I did test it manually, and it wasn't really my idea to add this, I thought you were concerned about this particular name collision on existing sites.